### PR TITLE
feat(payments): Android Billing SDK v6 updates

### PR DESCRIPTION
**Pull request by NathanWalker**
closes https://github.com/NativeScript/payments/issues/26

### CODE REVIEW

**Comment by healerjoy** [comment]
@NathanWalker As this code has not been released [here](https://www.npmjs.com/package/@nativescript/payments) yet, can you please share the procedure to build this into a runnable nativescript plugin. Need to use it immediately in a live nativescript project.

Thank you for your help @NathanWalker.

**Comment by healerjoy** [comment]
@NathanWalker Have got it working. Will complete the testing on the live app and share findings (if any).
Thank you for putting so much effort into NativeScript.

Anyone who would like to use this (before its merged into the main branch and published) can follow the below steps.

1. Download the [code from here](https://github.com/NativeScript/payments/tree/feat/payments-wallet-upgrade)
2. Unzip the code into yourfolder and "cd" into yourfolder
3. brew install gpg (Dependency for yarn. You can use any other method to install gpg as well)
4. asdf install yarn 1.22.19 (You can use any other method to install yarn as well)
5. npm run setup
6. npm run publish-packages
7. Then follow [these details](https://github.com/NativeScript/payments/tree/feat/payments-wallet-upgrade#how-to-publish-packages)
8. Use the generated files at dist/packages/payments as an npm package in your nativescript project.

**Comment by NathanWalker** [comment]
This is super helpful @healerjoy thank you. Yes only thing holding up merge and release is just doing more testing with it to ensure fully complaint. If you find it to be let us know here and we can publish the update upon your word.

**Comment by healerjoy** [comment]
@NathanWalker The entire payments cycle is currently working on the live app. To get the entire payments cycle to work where the product is a subscription product (billing library 6) - the following changes are necessary. If you give me access to the branch, I can make the changes and push to the same branch for your review.

1. [packages/payments/item/index.android.ts](https://github.com/NativeScript/payments/pull/27/files#diff-c89c5c9a5e8bb7aad6bbbdca2edcea328830120fe4a35ae4eb64a157c5677a42)
Lines 13 to 17 must be replaced with:
```js
this.type = nativeValue.getProductType();
if (this.type === com.android.billingclient.api.BillingClient.SkuType.INAPP) {
    const details = nativeValue.getOneTimePurchaseOfferDetails();
    this.priceAmount = details.getPriceAmountMicros() / 1000000;
    this.priceFormatted = details.getFormattedPrice();
    this.priceCurrencyCode = details.getPriceCurrencyCode();
} else if (this.type === com.android.billingclient.api.BillingClient.SkuType.SUBS) {
    const subscriptionOfferDetails = nativeValue.getSubscriptionOfferDetails().get(0)
    this.offerToken = subscriptionOfferDetails.getOfferToken();
    const details = subscriptionOfferDetails.getPricingPhases().getPricingPhaseList().get(0);
    this.priceAmount = details.getPriceAmountMicros() / 1000000;
    this.priceFormatted = details.getFormattedPrice();
    this.priceCurrencyCode = details.getPriceCurrencyCode();
}
```

2. [packages/payments/index.android.ts](https://github.com/NativeScript/payments/pull/27/files#diff-ed93466cee292bc8a5ff05166c37c16fb1dfdb91210b49675479360dae20ee4e)
 Lines 208 to 210 must be replaced with:
```js
let details
if(skuType == com.android.billingclient.api.BillingClient.SkuType.INAPP) {
    details = com.android.billingclient.api.BillingFlowParams.ProductDetailsParams.newBuilder()
    .setProductDetails(item.nativeValue)
    .build();
} else if (skuType == com.android.billingclient.api.BillingClient.SkuType.SUBS) {
    details = com.android.billingclient.api.BillingFlowParams.ProductDetailsParams.newBuilder()
    .setProductDetails(item.nativeValue)
    .setOfferToken(item.offerToken)
    .build();
}
```

**Comment by NathanWalker** [comment]
Thank you, should be able to now @healerjoy

File: packages/google-pay/index.android.ts [hunk 1]
```diff
@@ -130,79 +128,77 @@ export class GooglePayBtn extends View {
...
+          paymentMethod['tokenizationSpecification'] = tokenSpecification;
         }
 
-        paymentMethod.put('parameters', parameters);
-
-        paymentDataRequest.put('allowedPaymentMethods', new JSONArray().put(paymentMethod));
+        paymentMethod['parameters'] =  parameters;
 
-        paymentDataRequest.put('transactionInfo', this._getTransactionInfo(args));
+        paymentDataRequest['allowedPaymentMethods'] = paymentMethod;
```
Comment (msn444) [inline packages/google-pay/index.android.ts:197]: Needs to be
`paymentDataRequest['allowedPaymentMethods'] = [paymentMethod];`

**Comment by topclaudy** [comment]
Any plan to release this?

**Comment by NathanWalker** [comment]
@healerjoy let us know if you’d like to push the last adjustment or we can handle here to publish the major bump on Monday.

**Comment by healerjoy** [comment]
@cla-bot check

**Comment by healerjoy** [comment]
@NathanWalker I have pushed the update. Sorry that it was not done sooner. Request to please review and merge.

**Comment by topclaudy** [comment]
@healerjoy Thanks a lot!

**Comment by topclaudy** [comment]
When will it be released? I still get 1.0.6 and I can't find any tags/releases on this repo

**Comment by NathanWalker** [comment]
@nativescript/payments 2.0.0
@nativescript/google-pay 1.0.0

**Comment by NathanWalker** [comment]
@topclaudy Just fyi - In the future we will be integrating the forthcoming `nx release` workflow (announced today at NxConf) into all plugin repos to help auto generate changelog/release tagging to help see those releases better.

**Comment by topclaudy** [comment]
> @topclaudy Just fyi - In the future we will be integrating the forthcoming `nx release` workflow (announced today at NxConf) into all plugin repos to help auto generate changelog/release tagging to help see those releases better.

Nice stuff! Thanks for all the good work. Despite the bashing (people saying it's dead etc.), NativeScript is still my favorite for building mobile apps. Keep up the good work!!

**Comment by NathanWalker** [comment]
Hey thanks! We really enjoy developing with it so happy to help community anytime!

**Pull request merged**